File: src/com/nerdscorner/android/plugin/utils/MouseUtils.java
package com.nerdscorner.android.plugin.utils;

import java.awt.event.MouseAdapter;
import java.awt.event.MouseEvent;

public final class MouseUtils {

    public abstract static class DoubleClickAdapter extends MouseAdapter {
        public void mouseClicked(MouseEvent e) {
            if (e.getClickCount() == 2) {
                onDoubleClick();
            }
        }

        public abstract void onDoubleClick();
    }
}

File: src/com/nerdscorner/android/plugin/utils/MouseUtils.kt
package com.nerdscorner.android.plugin.utils

import java.awt.event.MouseAdapter
import java.awt.event.MouseEvent

class MouseUtils {

    abstract class DoubleClickAdapter : MouseAdapter() {
        override fun mouseClicked(e: MouseEvent) {
            if (e.clickCount == 2) {
                onDoubleClick()
            }
        }

        abstract fun onDoubleClick()
    }
}
